packages/common/src/is-empty.function.ts

```diff
@@ -38,10 +41,22 @@ function isEmptyString(input: string): input is '' {
   return Object.is(input.length, 0);
 }
 
+function isEmptySet(input: Set<unknown>): input is Set<never> {
```

Comment:
```suggestion
function isEmptySetOrMap(
  input: Set<unknown> | Map<unknown, unknown>
): input is Set<never> | Map<never, never> {
  return input.size === 0;
}
```
nit: не уверен что прям стоит, но проверки слишком похожи